lib/src/repo.rs

```diff
@@ -967,3 +967,24 @@ impl MutableRepo {
 #[derive(Debug, Copy, Clone, Error)]
 #[error("Cannot rewrite the root commit")]
 pub struct RewriteRootCommit;
+
+#[derive(Debug, Error)]
+#[error("Cannot access {path}")]
+pub struct PathError {
+    pub path: PathBuf,
+    #[source]
+    pub error: io::Error,
+}
+
+pub(crate) trait IoResultExt<T> {
+    fn path<P: AsRef<Path>>(self, path: P) -> Result<T, PathError>;
+}
+
+impl<T> IoResultExt<T> for io::Result<T> {
+    fn path<P: AsRef<Path>>(self, path: P) -> Result<T, PathError> {
+        self.map_err(|error| PathError {
+            path: path.as_ref().into(),
```

Comment:
I've generally avoided `into()` when there are clearer options. `to_path_buf()` and `to_owned()` are both available here, right? Is there some consensus in the community how to choose in cases like this?

Reply:
There aren't strong community norms here, but I think erring on the side of explicit makes sense. I generally don't think about it too hard when the context is otherwise clear, and it's a bit more convenient to not have to remember the most specialized version of a given operation, but there is inarguably some readability benefit.